### PR TITLE
Allows support users to amend deferred applications

### PR DESCRIPTION
**Pull request by JonF1**
## Context
Support was unable to amend contact details of a candidate who applied last cycle but deferred their application. This was because we prevent any updates of the application form from a previous cycle. 

## Changes proposed in this pull request
If the application choice status is offer deferred and the application is from a previous cycle then their details can now be amended without an error being raised

## Guidance to review
Try to amend a candidate's details who applied last recruitment cycle and deferred their application

## Link to Trello card

https://trello.com/c/WrZA7T1Z/4172-allow-support-users-to-amend-deferred-applications

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by JonF1** [comment]
[Allow support users to amend deferred applications](https://trello.com/c/WrZA7T1Z/4172-allow-support-users-to-amend-deferred-applications)

**Review by stevehook — COMMENTED** [review]
Looks good! One question inline.

File: app/models/application_form.rb [hunk 1]
```diff
@@ -478,4 +478,8 @@ def add_support_reference
   def previous_recruitment_cycle?
     RecruitmentCycle.current_year >= recruitment_cycle_year
   end
+
+  def deferred?
```
[the remark this answers is not in the document]
Comment (JonF1) [reply app/models/application_form.rb]: yep definitely, I've made the changes

**Review by stevehook — APPROVED** [review]
:+1:

**Pull request merged**